specification/quantum/resource-manager/Microsoft.Quantum/preview/2019-11-04-preview/quantum.json

```diff
@@ -0,0 +1,910 @@
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "description": "Creates or updates a workspace resource.",
+        "operationId": "Workspaces_CreateOrUpdate",
+        "x-ms-examples": {
+          "QuantumWorkspacesGet": {
+            "$ref": "./examples/quantumWorkspacesPut.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "name": "QuantumWorkspace",
+            "description": "Workspace details.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/QuantumWorkspace"
+            }
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Workspace create/update request accepted.",
+            "schema": {
+              "$ref": "#/definitions/QuantumWorkspace"
+            }
+          },
+          "200": {
+            "description": "Successful.",
+            "schema": {
+              "$ref": "#/definitions/QuantumWorkspace"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        }
+      },
+      "patch": {
+        "operationId": "Workspaces_Update",
+        "x-ms-examples": {
+          "QuantumWorkspacesGet": {
+            "$ref": "./examples/quantumWorkspacesPatch.json"
+          }
+        },
+        "description": "Updates a Workspace resource.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "name": "QuantumWorkspace",
+            "description": "Quantum workspace details.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/QuantumWorkspace"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
so a PATCH is always synchronous, i.e. not long running?

Reply:
Actually, I don't see PATCH defined in our RP. Not sure if this is supported automatically by RPSaaS. @haohaiyu, do you know?

Reply:
PATCH is used ONLY for RPAAS async call to update resource status.  We do not directly support customer to call PATCH now

Reply:
If PATCH is not callable by the customer via the management.azure.com endpoint, then it shouldn't be part of the swagger. (Swagger is customer-facing.)

---
In reply to: [552983621](https://github.com/Azure/azure-rest-api-specs/pull/11266#discussion_r552983621) [](ancestors = 552983621)

Reply:
Removed PATCH from the swagger since it's not customer facing.